File: englising-data/model/lyric.py
from sqlalchemy import Column, ForeignKey, Integer, String, Float, Boolean, TIMESTAMP, DECIMAL, Text, Date, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from model.base import Base

class Lyric(Base):
    __tablename__ = 'lyric'
    lyric_id = Column(Integer, primary_key=True)
    track_id = Column(Integer, ForeignKey('track.track_id'), nullable=False)
    start_time = Column(DECIMAL(10, 3), nullable=False)
    end_time = Column(DECIMAL(10, 3), nullable=False)
    en_text = Column(Text, nullable=False)
    kr_text = Column(Text, nullable=False)
    created_at = Column(TIMESTAMP, nullable=False, default=func.now())
    updated_at = Column(TIMESTAMP, onupdate=func.now())

    track = relationship("Track", back_populates="lyrics")

File: englising-be/src/main/java/org/englising/com/englisingbe/music/lyric/entity/Lyric.java
package org.englising.com.englisingbe.music.lyric.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.englising.com.englisingbe.music.album.entity.Album;
import org.englising.com.englisingbe.music.track.entity.Track;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.sql.Timestamp;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "lyric")
public class Lyric {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "lyric_id")
    private Long albumId;

    @ManyToOne
    @JoinColumn(name = "track_id", referencedColumnName = "track_id")
    private Track track;

    @Column(name = "start_time", precision = 10, scale = 3)
    private BigDecimal startTime;

    @Column(name = "end_time", precision = 10, scale = 3)
    private BigDecimal endTime;

    @Column(name = "en_text", length = 512)
    private String enText;

    @Column(name = "kr_text", length = 512)
    private String krText;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}
